R/utils.R

```diff
@@ -0,0 +1,27 @@
+hintr2_file <- function(...) {
+  system.file(..., package = "hintr2", mustWork = TRUE)
+}
+
+schema_root <- function() {
+  hintr2_file("schema")
+}
+
+scalar <- function(val) {
+  jsonlite::unbox(val)
+}
+
+scalar_null <- function() {
```

Comment:
We talked briefly about this but since then I've seen different responses return this `scalar_null` and some return `json_null` I wonder if there is a good reason for having these different return types to represent (I think) the same concept.

Reply:
yeah, I am very confused about having both here - `json_null()` seems like the one we want?

Reply:
Did a bit of digging. In `hintr` we have a function `json_verbatim` which we use to say something is raw JSON so the serialiser doesn't interpret it as a string. In particular the response getting the status of a model run returns us a `json_verbatim("null")` which then gets `scalar` class put on it giving us the same value as `json_null()` hence why it is in the tests. 

The `scalar_null()` is coming from `validate_baseline` created by a call to `scalar(NA)` in `hintr`.

So difference is just because of different types of return values in hintr 
```
Browse[1]> class(scalar(NA))
[1] "scalar"  "logical"
Browse[1]> class(scalar(hintr:::json_verbatim("null")))
[1] "scalar" "json"  
```

Both when serialized through pkgapi give us a json null
```
expect_equal(pkgapi:::to_json(json_null()), pkgapi:::to_json(scalar_null()))
```
Which I think is why we weren't seeing this in the testing of hintr.

We've used both here. The queue stuff returns `json_verbatim("null")` when it doesn't have data but most of the endpoints return `scalar(NA)` to indicate no data.

Reply:
Agreed we want to use `json_verbatim("null")` everywhere instead of `scalar(NA)` so I will create a PR for this hintr

Reply:
> Agreed we want to use `json_verbatim("null")` everywhere instead of `scalar(NA)` so I will create a PR for this hintr

See https://github.com/mrc-ide/hintr/pull/183